### ISSUE TITLE
SMTP setup and logging

### ISSUE THREAD

**Issue by sfeu** [monicahq/monica#1436]
I have no success in setting up the connection to a mail server. Currently I am using the docker image and my biggest problem is that i am not able to configure any helpful logging that could give me an idea what wents wrong or if at least the mail server gets contacted.

i tried:
APP_ENV = local
APP_DEBUG = true
inspected the logs created in storage and the docker logs

Currently i am trying to trigger emails with  the new email doubleopt in feature.

Anyone has for instance a running gmail  setup that i can give a tried (i also tried the usual 587 / smtp.gmail.com / tls setup with no success)

I also tried MAIL_DRIVER=log what unfortunately does not add anything to the logs in my case.

Is there any chance to enable some kind of logging?

Thanks,
Sebastian

**Comment by lpefferkorn** [comment]
Same issue for me, without logs I am unable to troubleshoot.

**Comment by dd86k** [comment]
Likewise. There is absolutely no documentation for anything that relates to mailing.

**Comment by gramakri** [comment]
Monica is just using lavarel framework for sending email. So, you can just look up docs in lavarel.

I was debugging this for the Monica app on Cloudron. Adding `MAIL_DRIVER=log` does put the emails in `storage/logs/lavarel.log`.

On Cloudron, we have a built-in SMTP server. The combination that works for us is:
```
# Mail credentials used to send emails from the application.
MAIL_DRIVER=smtp
MAIL_HOST=mail # this is the internal host name
MAIL_PORT=2525 # internal STARTTLS port
MAIL_USERNAME=${MAIL_SMTP_USERNAME}
MAIL_PASSWORD=${MAIL_SMTP_PASSWORD}
MAIL_ENCRYPTION= # Since it's internal, we don't use encryption on Cloudron
MAIL_FROM_ADDRESS=${MAIL_FROM}
MAIL_FROM_NAME=Monica
```

Hope that helps someone.

**Comment by samjaninf** [comment]
For linux servers the maillog is located at /var/log/maillog. that should give you any messages related to smtp deliveries and such.

**Comment by djaiss** [comment]
@sfeu did you find a workaround?

**Comment by tbirrell** [comment]
I am also encountering this issue. The reminders are being created, and then deleted when I run `send:reminders`, but there is no email. I've tried Mailtrap and MAIL_DRIVER=log. Nothing at all. No errors anywhere. The Unit Tests even pass. I am at a complete loss about what to try next. I'd be happy if I could get even the barest sort of logging so I can troubleshoot.

**Comment by samjaninf** [comment]
@tbirrell Which server are you running it on? Linux? Windows? Oracle? For CentOS/Fedora/Any Flavor RHEL the log to see if any email is actually being sent is /var/log/maillog, also check to make sure you have postfix/sendmail installed/running.

**Comment by tbirrell** [comment]
@samjaninf, Ubuntu on Digital Ocean. I can send email from the command line via `echo "test message" | sendmail test@mydomain.com`. But anything I send from Monica just fails silently. Nothing shows up in the log or the `.err` file

**Comment by samjaninf** [comment]
Are you getting any errors in Monica's log files? They should be located in /path/to/monica/storage/logs if I recall correctly.

**Comment by tbirrell** [comment]
Nope. Nothing. Its as if Laravel sends the reminder to the Queue, then just forgets about it. I'm currently investigating this line of thought right now.

**Comment by tbirrell** [comment]
For future readers, my issue was not an email problem (birthday notifications work), it is a reminders problem (still don't work). So if you are debugging, consider this as an option.

**Comment by djaiss** [comment]
Reminders are linked to the "paid" feature and if they are not sent, it might be linked to that. Are you sure you've set the `REQUIRES_SUBSCRIPTION` env variable to `false`?

**Comment by samjaninf** [comment]
My reminders are working currently. I just got a reminder today in my inbox. So I don't believe it is tied in that way.

**Comment by samjaninf** [comment]
My reminders are working currently. I just got a reminder today in my inbox. So I don't believe it is tied in that way. Do you have cron set up properly?

**Comment by mccollam** [comment]
I'm seeing similar behavior.  Things I've tried:

- gmail SMTP using TLS
- gmail SMTP using SSL
- sendmail using SMTP on localhost (confirmed to work outside of Monica)
- sendmail using `/usr/sbin/sendmail` (confirmed to work outside of Monica)
- `MAIL_DRIVER=log`

None of the above seems to work for either notifications or reminders.  Notifications at least process when the artisan job is run (as they are removed in the UI) but don't send mail.

Nothing ever appears in storage/logs or in the sendmail log (when using sendmail).

If there's a way to enable more verbose logging on this path I'm happy to do that and troubleshoot or provide more info!

**Comment by samjaninf** [comment]
Do you have your cronjob set up properly?

**Comment by mccollam** [comment]
@samjaninf if that was to me, then yes -- I can see that the cron job is running every minute.

But this shouldn't be affected by the cron job as I have run the same job by hand and observed that the notifications are being removed from the table (i.e. they are being processed) but nothing is being sent or logged (even when log output is set as the mail provider).

**Comment by mccollam** [comment]
I've done some more digging on this and managed to get some reminders sent out.  You must have the time to send reminders in your preferences set to the current hour if you are testing this, which is somewhat unexpected.

What I'd expect:
Running `php artisan send:reminders` would send all scheduled reminders if this is the right time to send them, or leave them queued if not.

What I observe:
Running `php artisan send:reminders` **does not send** any reminders if the current hour does not match the hour set for notifications in the user preferences, but **still deletes the reminder entry**!

Thus it appeared that everything except mail was working when I was running tests, but what was actually happening was that the notification was being silently deleted without ever attempting to send a message.

Not sure if that behavior is as expected or a bug but it was definitely confusing to me! :)  It also means that it is never safe to run `php artisan send:reminders` in production as it will delete users' reminders for the current day unless it is the time they would receive those reminders anyway.

**Comment by pwalker** [comment]
I'm also experiencing this issue.  My instance is installed on a linux host following the Generic instructions.   I've tried setting `APP_DEBUG=true` and `MAIL_DRIVER=log` and I can't get any new messages in my `.../storage/logs` folder.  I've tested the login credentials and security type in another email program and they work, but I don't think that should matter since I don't see anything showing up in the logs.

Any ideas?  I'm happy to help debug this further, but I'm not experienced at all with php/laravel so I don't know where to begin with this.

What I'd like to do is run a command like `php artisan send:test_notification 1234` where 1234 is my user id or something.  That would let me narrow it down to an email config/sending issue as opposed to a Reminder/Notification/Queue issue.

**Comment by DominikMa** [comment]
I had the same problem with my mail server. Found the answer here:
[laravel-expected-response-code-250-but-got-code-530](https://stackoverflow.com/questions/42574375/laravel-expected-response-code-250-but-got-code-530)

Had to set `MAIL_ENCRYPTION=tls` when using port 587 with STARTTLS.

**Comment by JC5** [comment]
Nothing so far seems to be working. I am also using the Docker instance and GMail. The birthday emails seem to work (next up is tomorrow so I'll keep you posted) but I'm pretty tired of stumbling around in the dark.

If somebody has some hints I am open to them!

**Comment by noharm1010** [comment]
Same problem here with reminder emails.
I'm running Monica 2.16.0 on a raspberry pi 4b with Ubuntu server 18.04.

Created a running server like mentioned in the generic instructions.
Everything works fine except the reminder e-mail.

I can create users, change password, invite users and I DO receive a e-mail.
But the reminders aren't send. Any suggestions?

**Comment by noharm1010** [comment]
with the use of [this post](https://github.com/monicahq/monica/issues/3324) "Reminders not working #3324" I made it work for myself.

Something to do with the cron job.

**Comment by iloveitaly** [comment]
I ran into this issue as well. It was caused by two issues for me:

* I had `QUEUE_CONNECTION=redis` but different have a separate queue process running. Use `QUEUE_CONNECTION=sync` instead
* My email configuration was silently failing on the mail providers side.

[Here's the docker config that worked for me](https://github.com/monicahq/docker/pull/49)
